apps/condo/domains/notification/adapters/redStoreAdapter.js

```diff
@@ -0,0 +1,157 @@
+        const pushContext = {}
+
+        tokens.forEach((pushToken) => {
+            const pushType = pushTypes[pushToken] || PUSH_TYPE_DEFAULT
+            const preparedData = prepareData(data, pushToken)
+            const pushData = {
+                token: pushToken,
+                data: preparedData,
+                notification,
+                appId: appIds[pushToken],
+                ...DEFAULT_PUSH_SETTINGS,
+            }
+
+            if (!APPS_WITH_DISABLED_NOTIFICATIONS.includes(pushData.appId)) notifications.push(pushData)
+
+            if (!pushContext[pushType]) pushContext[pushType] = pushData
+        })
+
+        return [notifications, pushContext]
+    }
+
+    /**
+     * Manages to send notification to all available pushTokens of the user.
+     * @param notification
+     * @param tokens
+     * @param data
+     * @param pushTypes
+     * @returns {Promise<null|(boolean|T|{state: string, error: *})[]>}
+     */
+    async sendNotification ({ notification, data, tokens, pushTypes, appIds } = {}, isVoIP = false) {
+        if (!tokens || isEmpty(tokens)) return [false, { error: 'No pushTokens available.' }]
+
+        const [notifications, pushContext] = RedStoreAdapter.prepareBatchData(notification, data, tokens, pushTypes, appIds, isVoIP)
+        let result
+
+        if (!isNull(this._config) && !isEmpty(notifications)) {
+            const notificationsByAppId = {}
+            for (const notification of notifications) {
+                const appId = notification.appId
+                notificationsByAppId[appId] ||= []
+                notificationsByAppId[appId].push(notification)
+            }
+            for (const [appId, notificationsBatchForApp] of Object.entries(notificationsByAppId)) {
+                const configForApp = this._config[appId]
+                if (!configForApp) {
+                    logger.error({ msg: 'Unknown appId. Config was not found', appId })
+                    continue
+                }
+
+                const app = new RedStoreNotificationSender(configForApp)
+                try {
+                    result = await app.sendAll(notificationsBatchForApp)
+
+                    if (!isEmpty(result.responses)) {
+                        result.responses = result.responses.map(
+                            (response, idx) =>
+                                ({
+                                    ...response,
+                                    pushToken: notifications[idx].token,
+                                    pushType: get(pushTypes, notifications[idx].token, null),
+                                })
+                        )
+                    }
+
+                } catch (err) {
+                    logger.error({ msg: 'sendNotification error', err })
+                }
+            }
+        } else {
+            throw new Error('config and notifications cannot be empty')
```

Comment:
Is it ok that you going to throw error at execution time for bad configuration? Probably better to know about bad configuration at app start time?

Reply:
this is necessary for extensibility. The constructor checks whether the config is empty or not. Next, we want to send push notifications to different applications, respectively, with different configs, and this should not require adding code to condo. Just update the config in Helm by adding a new application. Without knowing how many configs there should be, I cannot validate them up to this point

Reply:
So can you check if adapter is prepared for work before to call sendNotification? Like just add function isPrepared or something like that